### PR TITLE
Run Wasp CLI through cabal locally

### LINKED ISSUE #4178: Consider using `cabal run` instead of `wasp-cli` over the repository
Currently, we rely on `run install` to prepare the environment for most of our testing needs.

The problem is that on each change we have to re-run `run install` (or `cabal install` directly) so our tests can pull the changes. 
Personally I often lose time both to waiting for the `run install` command execution (or for executing the commands manually), and on forgetting I have to re-run `run install` before any test commands.

Using `cabal run` would make sure that our tests always work with the current state of the `waspc` project.

TODO: explore if there is an easy way to achieve this functionality

Question: Does using `cabal run` instead of `wasp-cli` makes us miss any edge-cases? Any different behavior?

### PR DESCRIPTION
**Pull request by FranjoMindek**
## Description

Fixes: https://github.com/wasp-lang/wasp/issues/4178

Adopts the same behavior we use in `playwright` tests to our `waspc` e2e-tests.
That is, we set the `WASP_CLI_CMD` through environment variable.

After that it:
- Changes the logic so that we run tests locally (`./run` script) through `cabal run`, which ensures we always use the current development version of Wasp.
- Keeps the same behavior for CI. 

## Checklist

- [ ] I tested my change in a Wasp app to verify that it works as intended.

- 🧪 Tests and apps:

  - [ ] I added **unit tests** for my change. 
  - [ ] _(if you fixed a bug)_ I added a **regression test** for the bug I fixed. 
  - [ ] _(if you added/updated a feature)_ I added/updated **e2e tests** in `examples/kitchen-sink/e2e-tests`.
  - [ ] _(if you added/updated a feature)_ I updated the **starter templates** in `waspc/data/Cli/templates`, as needed.
  - [ ] _(if you added/updated a feature)_ I updated the **example apps** in `examples/`, as needed.
    - [ ] _(if you updated `examples/tutorials`)_ I updated the tutorial in the docs (and vice versa).

- 📜 Documentation:

  - [ ] _(if you added/updated a feature)_ I **added/updated the documentation** in `web/docs/`.

- 🆕 Changelog: _(if change is more than just code/docs improvement)_
  - [ ] I updated `waspc/ChangeLog.md` with a **user-friendly** description of the change.
  - [ ] _(if you did a breaking change)_ I added a step to the current **migration guide** in `web/docs/migration-guides/`.
  - [ ] I **bumped the `version`** in `waspc/waspc.cabal` to reflect the changes I introduced.

### CODE REVIEW

File: waspc/e2e-tests/Main.hs [hunk 1]
```diff
@@ -32,7 +35,20 @@ main :: IO ()
 main = do
   if os == "mingw32"
     then putStrLn "Skipping end-to-end tests on Windows due to tests using *nix-only commands"
-    else e2eTests >>= defaultMain
+    else do
+      ensureE2eTestsEnvironment
+      e2eTests >>= defaultMain
+
+ensureE2eTestsEnvironment :: IO ()
```
Comment (FranjoMindek) [inline waspc/e2e-tests/Main.hs:43]: It is possible to evaluate the env value to a string, but the only neat way I found to do it is through  `unsafePerformIO` unless I missed something.

File: waspc/tools/wasp-cli-dev.ts
Comment (cprecioso) [inline waspc/tools/wasp-cli-dev.ts:1]: Feels like could be a shell script?

Comment (FranjoMindek) [reply waspc/tools/wasp-cli-dev.ts:1]: I thought we are moving away from those?
To get cross platform benefits eventually.

Comment (cprecioso) [reply waspc/tools/wasp-cli-dev.ts:1]: The binary is not written anywhere on disk?

**Review by cprecioso — APPROVED** [review]
The code looks fine. The solution doesn't completely sit fine with me, but you've already researched this so if you think it's the way to go, let's do it!

File: waspc/tools/wasp-cli-dev.ts
Comment (FranjoMindek) [reply waspc/tools/wasp-cli-dev.ts:1]: Note: when using binary directly, we miss the "forced compilation" part of `cabal run`. Meaning we can run an outdated binary.

**Comment by FranjoMindek** [comment]
> The code looks fine. The solution doesn't completely sit fine with me, but you've already researched this so if you think it's the way to go, let's do it!

TODO: potentially convert the TS script to bash if we go that way
